spec/services/slack_service_spec.rb

```diff
@@ -0,0 +1,236 @@
+        SlackServiceSpec::SlackClientMock
+      )
+      expect(result).to eql(:ok)
+      expect(data).to eql("Message Sent")
+
+      # The following asserts send_message receives th
+    end
+
+    it "failure: given a missing user - returns an error" do
+      result, data = SlackService.send_message("D00DD0DDD0D", "A message", SlackServiceSpec::SlackClientMock)
+      expect(result).to eql(:error)
+      expect(data).to eql("Slack Channel Not Found")
+    end
+  end
+
+  describe ".cobble_message_params" do
+    it "success: receives simple text messages" do
+      result = SlackService.cobble_message_params("The ID", "The Message")
+      expect(result).to eql({channel: "The ID", text: "The Message"})
+    end
+
+    it "success: receives complex text messages" do
+      result = SlackService.cobble_message_params("The ID", {
+        text: "A message",
+        attachments: [{"pretext": "pre-hello", "text": "text-world"}],
+        blocks: [{"type": "section", "text": {"type": "plain_text", "text": "Hello world"}}]
+      })
+      expect(result).to eql({
+        channel: "The ID",
+        text: "A message",
+        attachments: [{"pretext": "pre-hello", "text": "text-world"}],
+        blocks: [{"type": "section", "text": {"type": "plain_text", "text": "Hello world"}}]
+      })
+    end
+
+    it "failure: requires one of text, attachments, or blocks" do
+      expect { SlackService.cobble_message_params("The ID", {}) }.to raise_error(SlackService::MessageFormatError, "Message missing")
+    end
+
+    it "failure: rejects aditional keys" do
+      expect { SlackService.cobble_message_params("The ID", {text: "The Message", other: "other"}) }.to raise_error(SlackService::MessageFormatError, "Message format incorrect")
+    end
+  end
+end
+
+module SlackServiceSpec
+  class SlackClientMock
+    def self.chat_postMessage(params)
+      if params[:channel] == "A00AA1AAA"
+        usergroup_list = Slack::Messages::Message.new
+        usergroup_list[:ok] = true
+      else
+        raise Slack::Web::Api::Errors::ChannelNotFound.new("not_found")
+      end
+    end
+
+    def self.usergroups_list
+      usergroup_list = Slack::Messages::Message.new
+      usergroup_list[:ok] = true
+      usergroup_list[:usergroups] = [usergroup_mock()]
+      usergroup_list
+    end
+
+    def self.usergroups_users_list(params)
+      if params[:usergroup] == "A00AA0AAA0A"
+        users_list = Slack::Messages::Message.new
+        users_list[:ok] = true
+        users_list[:users] = ["A00AA1AAA", "A00AA1AAB"]
+        users_list
+      else
+        raise Slack::Web::Api::Errors::NoSuchSubteam.new("not_found")
+      end
+    end
+
+    def self.users_info(params)
+      if params[:user] == "A00AA1AAA"
+        user = Slack::Messages::Message.new
+        user[:ok] = true
+        user[:user] = user_mock()
+        user
+      else
+        raise Slack::Web::Api::Errors::UserNotFound.new("not_found")
+      end
+    end
+
+    def self.profile_mock
```

Comment:
I think we can all these data with rspec fixtures

Reply:
This data would be coming from the API not Activerecord.  I'm really only including everything so future devs can see what we're expecting from the API... but I could remove anything that we're not using for simplicity.  I could also just use VCR -- but then we'd need to set up users in Slack to match our fixtures for that to work.